### PR TITLE
Fetch API func completed

### PR DESCRIPTION
**Pull request by Oscar-Espinoza**
In this PR, refactored my code to use the Bookstore API to add, delete and get information, information is now persistent too.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -27,6 +38,6 @@ const booksSlice = createSlice({
   },
 });
 
-export const { addBook, removeBook } = booksSlice.actions;
+export const { addBook, removeBook, setBooks } = booksSlice.actions;
```
Comment (omar-muhamad) [inline src/redux/books/books.js]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/connec_to_api.md#:~:text=createAsyncThunk%20for%20making%20API%20calls.), Please use `createAsyncThunk `for creating thunk action creators.
[Hint] example for getting books data: 
```javascript
import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
); 
```

File: src/components/Book.jsx [hunk 2]
```diff
@@ -1,24 +1,31 @@
...
+import { baseUrl } from '../redux/store';
 
 function Book(props) {
   const dispatch = useDispatch();
 
+  const handleRemove = async (id) => {
+    await axios.delete(baseUrl + id)
+      .then(() => dispatch(removeBook(id)));
+  };
+
```
Comment (omar-muhamad) [inline src/components/Book.jsx]: - [ ] Same here, Please use createAsyncThunk for creating thunk remove action creator, and kindly move it to the books redux slice and import and use it here in `handleRemove` function.

File: src/components/NewBook.jsx [hunk 3]
```diff
@@ -1,18 +1,31 @@
...
+  const handleSubmit = async (e) => {
     e.preventDefault();
-    dispatch(addBook(formData));
+    await axios.post(baseUrl, formData)
+      .then(() => {
+        dispatch(addBook(formData));
+        setFormData({
+          ...initialState,
+          item_id: generateID(),
+        });
```
Comment (omar-muhamad) [inline src/components/NewBook.jsx]: - [ ] Same here, Please use createAsyncThunk for creating thunk add book action creator, and kindly move it to the books redux slice and import and use it here in handleSubmit function.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Oscar-Espinoza 

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by descholar-ceo — APPROVED** [review]
Hi @Oscar-Espinoza 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented all of the required changes from the previous review ✔️ 
👍🏽 Linters pass without any error ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**